src/lib/validation.js

```diff
@@ -39,21 +48,50 @@ const clearValidation = (document, collection) => {
   collection.set(document.uri, null);
 };
 
-const getRenderingErrors = async (source) => {
+const getRenderingErrors = async (source, context) => {
   try {
-    const { renderingErrors } = await validateHubl(getPortalId(), source);
+    const { renderingErrors } = await validateHubl(
+      getPortalId(),
+      source,
+      context
+    );
     return renderingErrors;
   } catch (e) {
     console.error('There was an error validating this file');
   }
 };
 
+const getTemplateType = (document) => {
+  if (isCodedFile(document.fileName)) {
+    const annotations = getAnnotationsFromSource(document.getText());
+    return {
+      is_available_for_new_content:
+        getAnnotationValue(
+          annotations,
+          ANNOTATION_KEYS.isAvailableForNewContent
+        ) != 'false',
```

Comment:
Code looks solid and makes sense, just wondering if we can improve the way that `getAnnotationValue` works to make usage a bit cleaner 🤔

Reply:
Yeah, I never really loved my implementation of that one. I just had one idea though, what if instead of `getAnnotationsFromSource`, we had something like `buildAnnotationValueGetter`. Basically the output would be a `getAnnotationValue` function that doesn't require the annotations to be passed into it. 
For example:

```js
const getAnnotationsFromSource = source => {
const annotations = //blah blah blah
return (...args) => getAnnotationValue(annotations, ...args);
}
```

Reply:
Thats a cool idea. Changed that up here https://github.com/HubSpot/hubspot-cli/pull/534/commits/42459f358c58d5c37108d007036bfdce4aa989eb